packages/jellyfish-json/src/index.ts

```diff
@@ -28,6 +35,61 @@ const reviveLosslessAs = (transformer: (string: string) => any) => {
   }
 }
 
+/**
+ * Revive lossless with keys a type
+ * @param text json string that is used to tranform to a json object
+ * @param precision PrecisionMapping is a key value pair to allow revive value type
+ * @returns jsonObject
+ */
+const reviveLosslessWithKeys = (text: string, precision: PrecisionMapping): any => {
+  return remapLosslessObj(precision, parse(text))
+}
+
+/**
+ * Remap lossless
+ * @param precision Precision - 'bignumber', 'number', 'lossless'
+ * @param losslessObj lossless json object
+ * @returns losslessObj
+ */
+const remapLosslessObj = (precision: PrecisionMapping, losslessObj: any): any => {
+  for (const k in precision) {
+    const value = losslessObj[k]
+    if (value !== undefined && value !== null) {
```

Comment:
I think you can rewrite it like this to reduce code complexity, (by returning/exiting early) Not sure how much complexity it will be reduced.

```ts
if (value === undefined && value === null) { 
  continue
}

const precisionType = precision[k] as Precision
if (typeof precisionType !== 'string') {
  remapLosslessObj(precisionType, losslessObj[k])
} else if (value instanceof LosslessNumber) {
  losslessObj[k] = revive(precisionType, value)
}
```